lib/charms/livy.py

```diff
@@ -0,0 +1,90 @@
+from glob import glob
+from path import Path
+from subprocess import CalledProcessError, call
+import os
+import jujuresources
+from charmhelpers.core import hookenv
+from charmhelpers.core import host
+from charmhelpers.core import unitdata
+from charmhelpers.fetch.archiveurl import ArchiveUrlFetchHandler
+from jujubigdata import utils
+
+
+# Main Livy class for callbacks
+class Livy(object):
+    def __init__(self, dist_config):
+        self.dist_config = dist_config
+        self.resources = {
+            'livy': 'livy-%s' % utils.cpu_arch(),
+        }
+        self.verify_resources = utils.verify_resources(*self.resources.values())
+
+    def is_installed(self):
+          return unitdata.kv().get('livy.installed')
```

Comment:
This was a copypasta that was carried forward from the old, non-layered charms.  It's moot now and can be dropped (it will automatically be handled by the existing state logic).